Fix for maven 3 warning

usmanismail opened the pull request:
Fix for maven warning 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-compiler-plugin is missing. @ line 38, column 21

Commit:
Fix for maven warning 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-compiler-plugin is missing. @ line 38, column 21

Commit:
Added support for multiple languages

Commit:
Added support for multiple languages

sergei-ivanov commented:
Can you please re-format the sources to the original author's code style?
Currently, it is impossible to tell from the diffs what has actually been changed.

sergei-ivanov commented on pom.xml:
```diff
@@ -34,6 +34,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-compiler-plugin</artifactId>
+                <version>2.3.2</version>
```
this was fixed in a bigger maven pom refactoring, please see pull request #10
https://github.com/dtrott/maven-protoc-plugin/pull/10

sergei-ivanov commented on src/main/java/com/google/protobuf/maven/Language.java:
```diff
@@ -0,0 +1,20 @@
+
+	public void setCompileCommand(String compileCommand) {
+		this.compileCommand = compileCommand;
+	}
```
enums are meant to be immutable, there is no need for a setter here, the constructor already does the job

sergei-ivanov commented on src/main/java/com/google/protobuf/maven/Language.java:
```diff
@@ -0,0 +1,20 @@
+public enum Language {
+	JAVA("--java_out="), CPP("--cpp_out=");
+
+	String compileCommand;
```
please make private

sergei-ivanov commented:
Despite all the formatting changes, I think I've got an idea of what you are proposing to do.

I think it can be achieved with very minimal changes to the plugin itself.
Basically, all we need is to inject a Language as a plugin property (with JAVA being a sensible default value).
Then the builder sets the protoc output according to the language setting (perhaps we need to rename Language to OutputType or something like that to avoid confusion with l10n/i18n stuff).

After all these changes, instead of looping through language specifications inside the plugin, the plugin user will define additional plugin executions per language, overriding the output type and the output directory for each execution.

Essentially, instead of making the plugin code and configuration more complicated, we can ask the plugin users to define additional plugin executions if needed.

sergei-ivanov commented:
Having thought more of it, and having reviewed pull request #15, I wonder if it'd make even more sense to split mojos per output type. Two mojos (for compile and test scopes) for Java, two for C++, two for Python. Then each mojo (or output-specific group of mojos) can define additional parameters and sensible defaults, instead of trying to cover all possible scenarios in a single mojo. Again, that will be more in a "Maven way" of doing things. And no 'language' parameter will be needed at all. All that's needed is some extension point in the AbstractProtocMojo, that would allow subclasses to add more configuration to the Protoc.Builder before finalising it.

Commit:
Allow for compiles without dependency resolution

I want to have the option to compile with either dependency resolution
or not. Make two new goals, that require dependency resolution, and
make the current ones not require it.

Commit:
remove eclipse crap

Commit:
remove target and add a .gitignore so that this doesn't happen again.

usmanismail committed:
Merge pull request #1 from niltz/master

not sure if you want this...

usmanismail committed:
Removing white space
